### PR TITLE
PCS/PBS Note - Transfer columns

### PR DESCRIPTION
**Pull request by NateHogan**

### CODE REVIEW

**Comment by meglenapnovoselska** [comment]
the screen with the 2 extra column looks a bit squeezed, would you mind giving equal space to all 4 columns?

also the second column (which is not part of your story) has a different name, do you know if that's fixed?

File: src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs [hunk 1]
```diff
@@ -57,6 +58,14 @@ public class ViewEvidenceNoteViewModel : EvidenceNoteViewModel
 
         public bool DisplayEvidenceNoteHistoryData => EvidenceNoteHistoryData != null && EvidenceNoteHistoryData.Count > 0;
 
+        public virtual IList<EvidenceCategoryValue> TransferCategoryValues { get; set; }
+
+        public bool DisplayTransferEvidenceColumns { get; set; }
+
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs:64]: unit tests for those

**Comment by meglenapnovoselska** [comment]
you still display 0.00 for cases when is null, it should be '-' for no values. check how the first two columns are displayed. they are correct.

File: src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs [hunk 2]
```diff
@@ -67,7 +67,9 @@ public EvidenceNoteData Map(EvidenceNoteWithCriteriaMap source)
                 history = transferredNotes.Select(n => new EvidenceNoteHistoryData(n.Id, n.Status.ToCoreEnumeration<NoteStatus>(), n.Reference, n.NoteType.ToCoreEnumeration<NoteType>(), 
                     n.NoteStatusHistory
                     .Where(n1 => n1.ToStatus.Equals(EA.Weee.Domain.Evidence.NoteStatus.Submitted))
-                    .OrderByDescending(n1 => n1.ChangedDate).FirstOrDefault()?.ChangedDate, n.Recipient.Scheme != null ? n.Recipient.Scheme.SchemeName : string.Empty)).OrderByDescending(n => n.Reference).ToList();
+                    .OrderByDescending(n1 => n1.ChangedDate).FirstOrDefault()?.ChangedDate, n.Recipient.Scheme != null ? n.Recipient.Scheme.SchemeName : string.Empty,
+                    source.Note.NoteTonnage.SelectMany(nt => nt.NoteTransferTonnage).Distinct().ToList().Where(x => x.TransferNote.Status.ToCoreEnumeration<NoteStatus>().Equals(NoteStatus.Approved) && x.TransferNoteId == n.Id)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs:71]: @NateHogan  Enhance the existing unit tests around these can do in another PR

File: src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs [hunk 3]
```diff
@@ -112,7 +113,29 @@ public ViewEvidenceNoteViewModel Map(ViewEvidenceNoteMapTransfer source)
...
+
+            if (TransferHistoryHasApprovedTransferNotes(source.EvidenceNoteData.EvidenceNoteHistoryData))
+            {
+                model.DisplayTransferEvidenceColumns = true;
+
+                for (var i = model.TransferCategoryValues.Count - 1; i >= 0; i--)
+                {
+                    var category = model.TransferCategoryValues.ElementAt(i);
+
+                    var transferTonnage = source.EvidenceNoteData.EvidenceNoteHistoryData.SelectMany(x => x.TransferEvidenceTonnageData).Where(y => y.CategoryId == (WeeeCategory)category.CategoryId).Distinct().ToList();
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/ViewModels/Shared/Mapping/ViewEvidenceNoteViewModelMap.cs:125]: @NateHogan Unit tests for these in another PR
